src/serac/numerics/functional/functional.hpp

```diff
@@ -288,49 +425,52 @@ class Functional<test(trial), exec> : public mfem::Operator {
    *
    * @note This gets more interesting when having more than one trial space
    */
-  void SetEssentialBC(const mfem::Array<int>& ess_attr)
+  void SetEssentialBC(const mfem::Array<int>& ess_attr, size_t which)
   {
-    static_assert(std::is_same_v<test, trial>, "can't specify essential bc on incompatible spaces");
-    trial_space_->GetEssentialTrueDofs(ess_attr, ess_tdof_list_);
+    // TODO check that it actually makes sense to apply bcs to this trial space
```

Comment:
We should have a discussion as to whether we want `Functional` to be essential BC aware at all. I'll add a discussion point to the meeting agenda.

Reply:
I'm leaning more and more toward moving essential BCs out of `serac::Functional`. They sort of make sense in the single-physics case (only when test and trial space match), but in all other cases (different test/trial spaces, multiple trial spaces, quantities of interest), it's unclear what it means to enforce boundary conditions.

Reply:
For the record, we discussed this and determined it was best to move the essential boundary condition operations outside of `Functional`.

Reply:
Do you think this change should be added to this PR, or come at a later date?

Reply:
`num_trial_spaces` will likely never be greater than 4 anyway, so I don't think there would any measurable difference in runtime